network/src/services/outbound_peer.rs

```diff
@@ -106,15 +108,19 @@ impl Future for OutboundPeerService {
                         let new_outbound = status
                             .max_outbound
                             .saturating_sub(status.non_whitelist_outbound);
-                        if self.network_state.config.whitelist_only {
-                            self.try_dial_whitelist()
-                        } else if new_outbound > 0 {
-                            // dial peers
-                            self.dial_peers(false, new_outbound as u32);
-                        } else {
-                            // feeler peers
-                            self.dial_peers(true, FEELER_CONNECTION_COUNT);
+                        if !self.network_state.config.whitelist_only {
+                            if new_outbound > 0 {
+                                // dial peers
+                                self.dial_peers(false, new_outbound as u32);
+                            } else {
+                                // feeler peers
+                                self.dial_peers(true, FEELER_CONNECTION_COUNT);
+                            }
```

Comment:
Seems the previous code is buggy(didn't dial_whitelist when `whitelist_only == false`)?

Reply:
`try_dial_whitelist` in all time

Reply:
Oh, I see. jjyr has removed `self.try_dial_whitelist();` from `dial_peers` https://github.com/nervosnetwork/ckb/pull/1413/files#diff-bb4e2e506c663eed7307ac58d00d7071L62.